### PR TITLE
Move generic code from nydusd into nydus crate

### PR DESCRIPTION
**Pull request by jiangliu**
Move generic code from nydusd into nydus crate, so they can be reused by other projects.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2,38 +2,143 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+//<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```
Comment (bergwolf) [inline src/lib.rs:5]: ??

**Review by bergwolf — COMMENTED** [review]
@jiangliu Could you explain the purpose of the PR? Which component is going to consume the nydus crate and if the nydus crate is a library crate, why not move it to the top level as other library crates?

**Comment by jiangliu** [comment]
> @jiangliu Could you explain the purpose of the PR? Which component is going to consume the nydus crate and if the nydus crate is a library crate, why not move it to the top level as other library crates?

The target audience is `image-rs` for confidential containers.
I reused `nydus` crate for simplicity. And it's Ok to split out as a dedicated library crate.

File: src/lib.rs (hunk 1 above)
Comment (jiangliu) [reply src/lib.rs:5]: A leftover, fixed

**Review by imeoer — APPROVED** [review]
LGTM

**Pull request merged**